cetmix_tower_server/models/cetmix_tower.py

```diff
@@ -99,3 +112,79 @@ def server_get_variable_value(
         if not value and check_global:
             value = result.get("global")
         return value
+
+    @api.model
+    def server_check_ssh_connection(self, server_reference, attempts=5, timeout=15):
+        """Check if SSH connection to the server is available.
+
+        Args:
+            server_reference (Char): Server reference.
+            attempts (int): Number of attempts to try the connection.
+                Default is 5.
+            timeout (int): Timeout in seconds for each connection attempt.
+                Default is 15 seconds.
+
+        Returns:
+            dict: {
+                "code": int,
+                    # 0 for success,
+                    # -1 for not found,
+                    # 1 for connection error,
+                    # 2 for retry limit exceeded
+                "message": str  # Description of the result
+            }
+        """
+        server = self.env["cx.tower.server"].get_by_reference(server_reference)
+        if not server:
+            return {
+                "code": SERVER_NOT_FOUND,
+                "message": _("Server not found"),
+            }
+
+        # Prepare connection details
+        ssh_host = server.ip_v4_address or server.ip_v6_address
+        ssh_port = server.ssh_port if server.ssh_port else 22
+        username = server.ssh_username
+        password = server.ssh_password
+
+        ssh_connection = SSH(
+            host=ssh_host,
+            port=ssh_port,
+            username=username,
+            password=password,
+            mode="p" if password else "k",
+            allow_agent=False,
+            timeout=timeout,
+        )
+
+        for attempt in range(attempts):
+            try:
+                _logger.info(f"Attempt to connect to {ssh_host}:{ssh_port}")
+                ssh_connection.connection()
+                return {
+                    "code": SUCCESS,
+                    "message": _(
+                        "SSH connection successful on attempt %d" % (attempt + 1)
+                    ),
+                }
+            except Exception as e:
+                _logger.error(f"Connection attempt {attempt + 1} failed: {str(e)}")
+
+                if attempt == 0:
+                    return {
+                        "code": SSH_CONNECTION_ERROR,
+                        "message": _("Connection error on first attempt: %s" % str(e)),
+                    }
+
+                if attempt < attempts - 1:
```

Comment:
You should save `attempts - 1` in a variable in order to not compute it every time within the cycle.